backend/courses/management/commands/recompute_topics.py

```diff
@@ -151,5 +152,30 @@ def handle(self, *args, **kwargs):
             assert (
                 min_semester in all_semesters()
             ), f"--min-semester={min_semester} is not a valid semester."
-
+        semesters = sorted(
+            [sem for sem in all_semesters() if not min_semester or sem >= min_semester]
+        )
         recompute_topics(min_semester, verbose=True, allow_null_parent_topic=bool(min_semester))
+        if semesters:
+            recompute_historical_semester_probabilities(
+                current_semester=semesters[-1], verbose=True
+            )
+
+
+def recompute_historical_semester_probabilities(current_semester, verbose=False):
+    """
+    Recomputes the historical probabilities for all topics.
+    """
+    if verbose:
+        print("Recomputing historical probabilities for all topics...")
+    topics = Topic.objects.all()
+    # Iterate over each Topic
+    for i, topic in tqdm(enumerate(topics)):
```

Comment:
should verbose flag apply to tqdm or not? Not sure what pattern we use elsewhere, but you can use disable option

https://stackoverflow.com/questions/37091673/silence-tqdms-output-while-running-tests-or-running-the-code-via-cron